examples/firewall/firewall.c

```diff
@@ -0,0 +1,373 @@
+                                        "Debug mode enabled; printing the source IP addresses of each incoming packet as well as drop/forward status\n");
+                                debug = 1;
+                                break;
+                        case '?':
+                                usage(progname);
+                                if (optopt == 'p')
+                                        RTE_LOG(INFO, APP, "Option -%c requires an argument.\n", optopt);
+                                if (optopt == 'd')
+                                        RTE_LOG(INFO, APP, "Option -%c requires an argument.\n", optopt);
+                                if (optopt == 'f')
+                                        RTE_LOG(INFO, APP, "Option -%c requires an argument.\n", optopt);
+                                else if (isprint(optopt))
+                                        RTE_LOG(INFO, APP, "Unknown option `-%c'.\n", optopt);
+                                else
+                                        RTE_LOG(INFO, APP, "Unknown option character `\\x%x'.\n", optopt);
+                                return -1;
+                        default:
+                                usage(progname);
+                                return -1;
+                }
+        }
+
+        if (!dst_flag) {
+                RTE_LOG(INFO, APP, "Firewall NF requires a destination NF with the -d flag.\n");
+                return -1;
+        }
+        if (!debug) {
+                RTE_LOG(INFO, APP, "Running normal mode, use -b flag to enable debug mode\n");
+        }
+        if (!rules_init) {
+                RTE_LOG(INFO, APP, "Please specify a rules JSON file with -f FILE_NAME\n");
+                return -1;
+        }
+        return optind;
+}
+
+/*
+ * This function displays stats. It uses ANSI terminal codes to clear
+ * screen when called. It is called from a single non-master
+ * thread in the server process, when the process is run with more
+ * than one lcore enabled.
+ */
+static void
+do_stats_display(void) {
+        const char clr[] = {27, '[', '2', 'J', '\0'};
+        const char topLeft[] = {27, '[', '1', ';', '1', 'H', '\0'};
+
+        /* Clear screen and move to top left */
+        printf("%s%s", clr, topLeft);
+        printf("Packets Dropped: %lu\n", stats.pkt_drop);
+        printf("Packets not IPv4: %lu\n", stats.pkt_not_ipv4);
+        printf("Packets Accepted: %lu\n", stats.pkt_accept);
+        printf("Packets Total: %lu", stats.pkt_total);
+
+        printf("\n\n");
+}
+
+static int
+packet_handler(struct rte_mbuf *pkt, struct onvm_pkt_meta *meta, __attribute__((unused)) struct onvm_nf_info *nf_info) {
+        struct ipv4_hdr *ipv4_hdr;
+        static uint32_t counter = 0;
+        int ret;
+        uint32_t rule = 0;
+        uint32_t track_ip = 0;
+
+        if (++counter == print_delay) {
+                do_stats_display();
+                counter = 0;
+        }
+
```

Comment:
As `stats.pkt_total++;` happens all the time it might be cleaner to just have it here